src/core/parser.js

```diff
@@ -846,7 +846,11 @@ var Lexer = (function LexerClosure() {
             }
             strBuf.push(String.fromCharCode((x << 4) | x2));
           } else {
-            strBuf.push('#', String.fromCharCode(ch));
+            strBuf.push('#');
+            if (!Lexer.isSpace(ch)) {
+              strBuf.push(String.fromCharCode(ch));
+            }
+            this.stream.skip(-1); // Reset the stream position.
```

Comment:
That sounds better, thank you pointing this out!
I've pushed a new version of the patch, is this what you had in mind?